pkg/parser/parser.go

```diff
@@ -0,0 +1,108 @@
+package parser
+
+import (
+	"fmt"
+
+	"github.com/tjgurwara99/monkey-go/pkg/ast"
+	"github.com/tjgurwara99/monkey-go/pkg/token"
+)
+
+type Parser struct {
+	tokens chan token.Token
+
+	current token.Token
+	peek    token.Token
+
+	errors []string
```

Comment:
Change this to something like this maybe - this doesn't change a lot but its idiomatic to keep `errors` to be types that satisfy `error` interface.

```suggestion
	errors []error
```